From 024157b96eede4558ab9feef2afced67db6e52ce Mon Sep 17 00:00:00 2001
From: LexManos <LexManos@gmail.com>
Date: Wed, 17 Jun 2020 11:07:09 -0700
Subject: [PATCH] Do not rebuild variable names in lambdas. Breaks outer this
 references. Code existed before my time, no idea what it's intention is.


diff --git a/src/org/jetbrains/java/decompiler/main/rels/NestedClassProcessor.java b/src/org/jetbrains/java/decompiler/main/rels/NestedClassProcessor.java
index 7f80109..fe37266 100644
--- a/src/org/jetbrains/java/decompiler/main/rels/NestedClassProcessor.java
+++ b/src/org/jetbrains/java/decompiler/main/rels/NestedClassProcessor.java
@@ -285,7 +285,7 @@ public class NestedClassProcessor {
     Set<String> setNewOuterNames = new HashSet<>(mapNewNames.values());
     setNewOuterNames.removeAll(method.setOuterVarNames);
 
-    method.varproc.refreshVarNames(new VarNamesCollector(setNewOuterNames));
+    //method.varproc.refreshVarNames(new VarNamesCollector(setNewOuterNames));
     method.setOuterVarNames.addAll(setNewOuterNames);
 
     for (Entry<VarVersionPair, String> entry : mapNewNames.entrySet()) {
-- 
2.27.0.windows.1

